File: core/BestTimeToBuyAndSellStockII/BestTimeToBuyAndSellStockII.py
from typing import List


def maxProfit(prices: List[int]) -> int:
    sell, buy, days = 0, -prices[0], len(prices)
    for day in range(days):
        temp = sell
        sell = max(sell, buy + prices[day])
        buy = max(buy, temp - prices[day])

    return sell

File: core/BestTimeToBuyAndSellStockII/BestTimeToBuyAndSellStockII.java
package com.dbc;

public class BestTimeToBuyAndSellStockII {
    public int maxProfit(int[] prices) {
        int sell = 0, buy = -prices[0];
        for (int price : prices) {
            int temp = sell;
            sell = Math.max(sell, buy + price);
            buy = Math.max(buy, temp - price);
        }
        return sell;
    }
}
